src/lem/circuit.rs

```diff
@@ -708,13 +734,85 @@ impl Func {
+                        let b_num = b.hash();
+                        let c_num = add(&mut cs.namespace(|| "add"), a_num, b_num)?;
+                        let tag = g
+                            .global_allocator
+                            .get_or_alloc_const(cs, Tag::Expr(Num).to_field())?;
+                        let c = AllocatedPtr::from_parts(tag, c_num);
+                        bound_allocations.insert(tgt.clone(), c);
+                    }
+                    Op::Sub(tgt, a, b) => {
+                        let a = bound_allocations.get(a)?;
+                        let b = bound_allocations.get(b)?;
+                        // TODO check that the tags are correct
+                        let a_num = a.hash();
+                        let b_num = b.hash();
+                        let c_num = sub(&mut cs.namespace(|| "sub"), a_num, b_num)?;
+                        let tag = g
+                            .global_allocator
+                            .get_or_alloc_const(cs, Tag::Expr(Num).to_field())?;
+                        let c = AllocatedPtr::from_parts(tag, c_num);
+                        bound_allocations.insert(tgt.clone(), c);
+                    }
+                    Op::Mul(tgt, a, b) => {
+                        let a = bound_allocations.get(a)?;
+                        let b = bound_allocations.get(b)?;
+                        // TODO check that the tags are correct
+                        let a_num = a.hash();
+                        let b_num = b.hash();
+                        let c_num = mul(&mut cs.namespace(|| "mul"), a_num, b_num)?;
+                        let tag = g
+                            .global_allocator
+                            .get_or_alloc_const(cs, Tag::Expr(Num).to_field())?;
+                        let c = AllocatedPtr::from_parts(tag, c_num);
+                        bound_allocations.insert(tgt.clone(), c);
+                    }
+                    Op::Div(_tgt, _a, _b) => {
+                        // TODO
+                    }
+                    Op::Hide(tgt, _sec, _pay) => {
+                        // TODO
                         let allocated_ptr = AllocatedPtr::from_parts(
-                            g.global_allocator.get_or_alloc_const(cs, tag.to_field())?,
+                            g.global_allocator.get_or_alloc_const(cs, F::ZERO)?,
                             g.global_allocator.get_or_alloc_const(cs, F::ZERO)?,
                         );
                         bound_allocations.insert(tgt.clone(), allocated_ptr);
                     }
-                    _ => todo!(),
+                    Op::Open(pay, sec, _comm_or_num) => {
+                        // TODO
+                        let allocated_ptr = AllocatedPtr::from_parts(
+                            g.global_allocator.get_or_alloc_const(cs, F::ZERO)?,
+                            g.global_allocator.get_or_alloc_const(cs, F::ZERO)?,
+                        );
```

Comment:
I think this is meant to be stand-in for actually getting the opening from the `Store`. If you can replace this with `todo!()` and not include 'definitely wrong code' unadorned, that would be better.

Reply:
I'll just make it empty for now, with a todo comment, so it does not make eval fail